### ISSUE TITLE
Showing second marriages in a sandclock

### ISSUE THREAD

**Issue by alexreg** [T-F-S/genealogytree#54]
Is it possible to show second marriages among ancestors of a given individual that is the centre of a sandclock? It seems that only `child` blocks can contain `union` blocks, not `parent` blocks, which I would need for this purpose. (I see no theoretical obstacle to `parent` blocks also supporting `union` blocks inside.)

**Comment by T-F-S** [comment]
No, I fear `union` blocks are only supported in `child` blocks.

**Comment by alexreg** [comment]
@T-F-S Yes, it seems so, right now. But it would be super useful to have support in `parent` blocks too. I see no theoretical reason why not, so it should be "just" a matter of someone making the effort to implement it, no?

**Comment by T-F-S** [comment]
*Genealogy trees*, as denoted by the package, are rooted ordered trees with some annotations. A *family* is an ordered set of parent nodes or child nodes. One of these nodes is a g-node which is parent to one family and child to another family.

In `child` type families, the `g` node is a parent of the family. Its spouse is another parent of the family (annotated). `union`s are tricked in to add further parents and child. In the end, the *family* becomes a patchwork family with one `g` node, severals `p` nodes (seen as annotation) and multiple `child`s (subtrees) or `c` nodes (annotations). Nevertheless, the *family* stays an ordered set of parent nodes or child nodes. Interconnecting lines are an additional complication.

In `parent` type families, the `g` node is a child of the family. Its spouse is not part of the family. Other (annotated) `c` nodes are siblings of the `g` node. Also there are multiple `parent`s (subtrees) or `p` nodes (annotations). There would be no problem to have more the 2 parents for the `g` node from the theoretical side; it is just a biology resctricts. But, `union`s would be structural different here, because they would add further nodes on the layer of the `g` node where only siblings are present. Even, if one would discriminate between sibling nodes and alien nodes, there would remain the fact, that the (primary) spouse is not part of the family (and layout), but union spouses would be. Here, a *family* would be become a set of parent nodes, a set of child nodes and another set of spouse nodes, but not containing the primary spouse.

I hope, this answer helps to understand that a `union` is structural different for `parent` and `child` subgraphs (and therefore only implemented for `child`).

**Comment by alexreg** [comment]
@T-F-S Thank you for the clear explanation. Unfortunately, then, it seems impossible right now to show multiple marriages in an "ancestor tree". I appreciate this transforms an "ancestor tree" (in the strict sense) into something more general, though I can imagine this would be a highly desirable feature in various contexts; it certainly would be for me. If it indeed this not supported right now, would you have any suggestions on how to implement it under your current design? This would either be for a personal fork of the package, or a pull request, if you're interested. Thanks.

**Comment by T-F-S** [comment]
> If it indeed this not supported right now, would you have any suggestions on how to implement it under your current design?

Sorry, but I think the implementation is not easy or maybe impossible with the current model. I would like to have this feature for my own projects as well...

**Comment by alexreg** [comment]
@T-F-S I presume that even though this issue is closed "as completed", it's not actually been implemented yet?

**Comment by T-F-S** [comment]
Yes, you are right. I still think that an implementation with the current model is not possible in a senseful way.

Thank you for hinting me to different `closed` options (seems, I overlooked that).
